cirq/qis/measures_test.py

```diff
@@ -30,40 +30,45 @@ def test_fidelity_symmetric():
                                cirq.fidelity(VEC2, VEC1))
     np.testing.assert_allclose(cirq.fidelity(VEC1, MAT1),
                                cirq.fidelity(MAT1, VEC1))
-    np.testing.assert_allclose(cirq.fidelity(MAT1, MAT2),
-                               cirq.fidelity(MAT2, MAT1))
+    np.testing.assert_allclose(
+        cirq.fidelity(cirq.density_matrix(MAT1, dtype=np.complex128), MAT2),
+        cirq.fidelity(cirq.density_matrix(MAT2, dtype=np.complex128), MAT1))
 
 
 def test_fidelity_between_zero_and_one():
     assert 0 <= cirq.fidelity(VEC1, VEC2) <= 1
     assert 0 <= cirq.fidelity(VEC1, MAT1) <= 1
-    assert 0 <= cirq.fidelity(MAT1, MAT2) <= 1
+    assert 0 <= cirq.fidelity(cirq.density_matrix(MAT1, dtype=np.complex128),
+                              MAT2) <= 1
 
 
 def test_fidelity_invariant_under_unitary_transformation():
     np.testing.assert_allclose(
-        cirq.fidelity(MAT1, MAT2),
-        cirq.fidelity(U @ MAT1 @ U.T.conj(), U @ MAT2 @ U.T.conj()))
+        cirq.fidelity(cirq.density_matrix(MAT1, dtype=np.complex128), MAT2),
+        cirq.fidelity(
+            cirq.density_matrix(U @ MAT1 @ U.T.conj(), dtype=np.complex128),
+            U @ MAT2 @ U.T.conj()))
 
 
 def test_fidelity_commuting_matrices():
     d1 = np.random.uniform(size=N)
     d1 /= np.sum(d1)
     d2 = np.random.uniform(size=N)
     d2 /= np.sum(d2)
-    mat1 = U @ np.diag(d1) @ U.T.conj()
+    mat1 = cirq.density_matrix(U @ np.diag(d1) @ U.T.conj(),
+                               dtype=np.complex128)
     mat2 = U @ np.diag(d2) @ U.T.conj()
 
     np.testing.assert_allclose(cirq.fidelity(mat1, mat2),
                                np.sum(np.sqrt(d1 * d2))**2)
 
 
 def test_fidelity_known_values():
-    vec1 = np.array([1, 1j, -1, -1j]) * 0.5
-    vec2 = np.array([1, -1, 1, -1]) * 0.5
-    mat1 = np.outer(vec1, vec1.conj())
-    mat2 = np.outer(vec2, vec2.conj())
-    mat3 = 0.3 * mat1 + 0.7 * mat2
+    vec1 = np.array([1, 1j, -1, -1j], dtype=np.complex128) * 0.5
```

Comment:
I'm surprised we need to specify dtype here. Do you know why?

Reply:
Yeah, it wasn't necessary on this line. I'm actually considering removing the default dtype specification of complex64 in QuantumState and related methods, and instead by default reading the dtype from the numpy array passed in. I think that's more in line with the change proposed at https://github.com/quantumlib/Cirq/issues/3394.

Reply:
I've removed unnecessary dtype specifications and made specifying the dtype optional, following #3428 .